src/viewer.js

```diff
@@ -998,12 +1001,16 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
         // Go through top element (passed to us) and remove all children
         // Use removeChild to make sure it handles SVG or any non-html
         // also it performs better - http://jsperf.com/innerhtml-vs-removechild/15
-        if (this.element){
-            while (this.element.firstChild) {
-                this.element.removeChild(this.element.firstChild);
+        if (this.container && this.canvas) {
+            if (this.canvas.parentNode === this.container) {
```

Comment:
I don't think we need to explicitly remove the canvas from the container. Removing the container from its parent should be enough.